src/v/config/configuration.cc

```diff
@@ -131,6 +131,14 @@ configuration::configuration()
       {.example = "65536"},
       std::nullopt,
       {.min = 32_KiB, .align = 4_KiB})
+  , rpc_client_connections_per_peer(
```

Comment:
Don't think it's needed but would this work with needs_restart=no?

Reply:
No,the changes in this PR don't support dynamically deleting or adding new connections to a broker. For the full implementation 23.3.x it won't need a restart.